File: Plugins/Aspose_Imaging_Java_for_Ruby/lib/asposeimagingjava/images/resizeimage.rb
module Asposeimagingjava
  module ResizeImage
    def initialize()
        # Simple Resizing
        simple_image_resizing()

        # Resizing with ResizeType Enumeration
        resizing_with_resizetype_enumeration()
    end

    def simple_image_resizing()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Cache data if not cached previously
        if !image.isCached()
            image.cacheData()
        end

        # Specify only width
        new_width = image.getWidth() / 2
        image.resizeWidthProportionally(new_width)

        # Specify only height
        new_height = image.getHeight() / 2
        image.resizeHeightProportionally(new_height)

        # Save the image to disk
        image.save(data_dir + "simple_image_resizing.jpg")

        # Display Status.
        puts "Resized image successfully!"
    end

    def resizing_with_resizetype_enumeration()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Cache data if not cached previously
        if !image.isCached()
            image.cacheData()
        end

        # Specify only width
        new_width = image.getWidth() / 2
        image.resizeWidthProportionally(new_width, Rjb::import('com.aspose.imaging.ResizeType').LanczosResample)

        # Specify only height
        new_height = image.getHeight() / 2
        image.resizeHeightProportionally(new_height, Rjb::import('com.aspose.imaging.ResizeType').NearestNeighbourResample)

        # Save the image to disk
        image.save(data_dir + "resizing_with_resizetype_enumeration.jpg")

        # Display Status.
        puts "Resized image successfully!"
    end
  end
end

File: Plugins/Aspose_Imaging_Java_for_PHP/src/aspose/imaging/ManagingRasterFormats/ResizeImage.php
<?php
namespace Aspose\Imaging\ManagingRasterFormats;

use com\aspose\imaging\Image as Image;
use com\aspose\imaging\ResizeType as ResizeType;
class ResizeImage{

    public static function run($dataDir=null){

        # Simple Resizing
        ResizeImage::simple_image_resizing($dataDir);

        # Resizing with ResizeType Enumeration
        ResizeImage::resizing_with_resizetype_enumeration($dataDir);
    }

    public static function simple_image_resizing($dataDir=null){

    # Load an existing image
    $image=new Image();
    $image = $image->load($dataDir . "test.jpg");

    # Cache data if not cached previously
    if (!$image->isCached()) {
    $image->cacheData();
    }

    # Specify only width
    $new_width = $image->getWidth()/2;
    $image->resizeWidthProportionally($new_width);

    # Specify only height
    $new_height = $image->getHeight()/2;
    $image->resizeHeightProportionally($new_height);

    # Save the image to disk
    $image->save($dataDir . "simple_image_resizing.jpg");

    # Display Status.
    print "Resized image successfully!".PHP_EOL;
    }

    public static function resizing_with_resizetype_enumeration($dataDir=null){

    # Load an existing image
    $image=new Image();
    $image = $image->load($dataDir . "test.jpg");

    # Cache data if not cached previously
    if (!$image->isCached()) {
    $image->cacheData();
    }

    # Specify only width
    $new_width = $image->getWidth()/2;
    $resizeType=new ResizeType();
    $image->resizeWidthProportionally($new_width, $resizeType->LanczosResample);

    # Specify only height
    $new_height = $image->getHeight()/2;
    $image->resizeHeightProportionally($new_height, $resizeType->NearestNeighbourResample);

    # Save the image to disk
    $image->save($dataDir . "resizing_with_resizetype_enumeration.jpg");

    # Display Status.
    print "Resized image successfully!".PHP_EOL;
    }
}
?>